cumulus/client/relay-chain-rpc-interface/src/light_client_worker.rs

```diff
@@ -124,7 +124,7 @@ pub struct LightClientRpcWorker {
 }
 
 fn handle_notification(
-	maybe_header: Option<Result<RelayHeader, Error>>,
+	maybe_header: Option<Result<RelayHeader, serde_json::Error>>,
```

Comment:
//cc @skunert 

This is slightly modified and only decoding can cause this error now.
Since this API was confusing before i.e, the subscription could be closed because it lagged (i.e, couldn't keep with server) or just closed.

We have added another API https://docs.rs/jsonrpsee-core/0.23.1/jsonrpsee_core/client/struct.Subscription.html#method.close_reason but not sure whether you care here....